autogpt_platform/backend/backend/server/integrations/router.py

```diff
@@ -69,6 +70,9 @@ def callback(
 ) -> CredentialsMetaResponse:
     logger.debug(f"Received OAuth callback for provider: {provider}")
     handler = _get_provider_oauth_handler(request, provider)
+    code_verifier = creds_manager.store._get_code_verifier(
+        user_id, provider, state_token
+    )
 
     # Verify the state token
     if not creds_manager.store.verify_state_token(user_id, state_token, provider):
```

Comment:
Could you please make `verify_state_token` return the `OAuthState` object and use that instead of the two separate `_get_code_verifier` and `get_any_valid_scopes_from_state_token` calls?